src/library_wasmfs_opfs.js

```diff
@@ -286,11 +286,14 @@ mergeInto(LibraryManager.library, {
 
   _wasmfs_opfs_close_access__deps: ['$wasmfsOPFSFree',
                                     '$wasmfsOPFSAccessHandles'],
-  _wasmfs_opfs_close_access: async function(ctx, accessID) {
+  _wasmfs_opfs_close_access: async function(ctx, accessID, errPtr) {
     let accessHandle = wasmfsOPFSAccessHandles.get(accessID);
-    // Wait for the close to finish to ensure that subsequent opens succeed. The
-    // close cannot fail, so we don't need any kind of error handling.
-    await accessHandle.close();
+    try {
+      await accessHandle.close();
+    } catch {
+      let err = -{{{ cDefine('EIO') }}};
```

Comment:
I guess this can't ever be e.g.  EBADF?

Reply:
Right, EBADF in particular should have been caught well before we get to this point. It would be good to report more specific error codes from these methods where possible, but that would require a deeper understanding of what different errors they can produce in the first place.

Reply:
based on my reading of `man 2 close` EBADF is the only other possible thing listed there aside from EIO that could make sense for us anyway.